cedar-policy/src/ffi/is_authorized.rs

```diff
@@ -444,18 +446,18 @@ pub enum PartialAuthorizationAnswer {
 pub struct AuthorizationCall {
     /// The principal taking action
     #[cfg_attr(feature = "wasm", tsify(type = "string|{type: string, id: string}"))]
-    principal: Option<JsonValueWithNoDuplicateKeys>,
+    principal: Option<serde_json::Value>,
```

Comment:
Not sure I like this. We went through some effort previously to catch this duplicate key case, and I think that was the right thing to do.

Reply:
My thinking was that we want the FFI to use simple types where possible, and users can decide how to handle duplicates when they construct the `serde_json::Value`.

The other option is to move `JsonValueWithNoDuplicateKeys` from core to the public API -- would you prefer that?

Reply:
IIRC, we decided downstream users should handle duplicates as they see fit. So in principle I'm fine with moving `JsonValueWithNoDuplicateKeys` to the public API. I'm not sure this will help anyone using the FFI though because they'll still likely use an equivalent of `serde_json::Value`on the other side of the interface.